streams/test-utils/src/test/java/org/apache/kafka/streams/MockProcessorContextTest.java

```diff
@@ -0,0 +1,410 @@
+
+import java.io.File;
+import java.util.Iterator;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class MockProcessorContextTest {
+    /**
+     * Behavioral test demonstrating the use of the context for capturing forwarded values
+     */
+    @Test
+    public void shouldCaptureOutputRecords() {
+        final AbstractProcessor<String, Long> processor = new AbstractProcessor<String, Long>() {
+            @Override
+            public void process(final String key, final Long value) {
+                this.context().forward(key + value, key.length() + value);
+            }
+        };
+
+        final MockProcessorContext context = new MockProcessorContext();
+
+        processor.init(context);
+
+        processor.process("foo", 5L);
+        processor.process("barbaz", 50L);
+
+        final Iterator<CapturedForward> forwarded = context.forwarded().iterator();
+        assertEquals(forwarded.next().kv(), new KeyValue<>("foo5", 8L));
+        assertEquals(forwarded.next().kv(), new KeyValue<>("barbaz50", 56L));
+        Assert.assertFalse(forwarded.hasNext());
+
+        context.resetForwards();
+
+        assertEquals(context.forwarded().size(), 0);
+    }
+
+    /**
+     * Behavioral test demonstrating the use of the context for capturing forwarded values using the To API
+     */
+    @Test
+    public void shouldCaptureOutputRecordsUsingTo() {
+        final AbstractProcessor<String, Long> processor = new AbstractProcessor<String, Long>() {
+            @Override
+            public void process(final String key, final Long value) {
+                this.context().forward(key + value, key.length() + value, To.all());
+            }
+        };
+
+        final MockProcessorContext context = new MockProcessorContext();
+
+        processor.init(context);
+
+        processor.process("foo", 5L);
+        processor.process("barbaz", 50L);
+
+        final Iterator<CapturedForward> forwarded = context.forwarded().iterator();
+        assertEquals(forwarded.next().kv(), new KeyValue<>("foo5", 8L));
+        assertEquals(forwarded.next().kv(), new KeyValue<>("barbaz50", 56L));
+        Assert.assertFalse(forwarded.hasNext());
+
+        context.resetForwards();
+
+        assertEquals(context.forwarded().size(), 0);
+    }
+
+    /**
+     * Behavioral test demonstrating the use of the context for capturing forwarded values to specific children
+     */
+    @Test
+    public void shouldCaptureRecordsOutputToChildByName() {
+        final AbstractProcessor<String, Long> processor = new AbstractProcessor<String, Long>() {
+            private int count = 0;
+
+            @Override
+            public void process(final String key, final Long value) {
+                if (count == 0) {
+                    this.context().forward("start", -1L, To.all()); // broadcast
```

Comment:
the broadcast was tested above already -- we should remove it -- one test for one feature

Reply:
Here, I'm verifying that it works in conjunction with send-to-child-by-name.